ndc-reference/bin/reference/main.rs

```diff
@@ -525,6 +608,140 @@ fn get_collection_by_name(
                 latest_id_value,
             )])])
         }
+        "latest_article" => {
+            let latest_id = state
+                .articles
+                .iter()
+                .filter_map(|(_id, a)| a.get("id").and_then(|v| v.as_i64()))
+                .max();
+
+            if let Some(id) = latest_id {
+                let latest_article = state.articles.get(&id);
+
+                let rows = query
+                    .fields
+                    .as_ref()
+                    .map(|fields| {
+                        let mut rows: Vec<IndexMap<String, models::RowFieldValue>> = vec![];
+                        for item in latest_article.iter() {
+                            let mut row = IndexMap::new();
+                            for (field_name, field) in fields.iter() {
+                                row.insert(
+                                    field_name.clone(),
+                                    eval_field(
+                                        &BTreeMap::new(),
+                                        &BTreeMap::new(),
+                                        state,
+                                        field,
+                                        item,
+                                    )?,
+                                );
+                            }
+                            rows.push(row)
+                        }
+                        Ok::<_, StatusLine>(rows)
+                    })
+                    .transpose()?;
+
+                let row_set = models::RowSet {
+                    aggregates: None,
+                    rows,
+                };
+
+                let latest_article_value = serde_json::to_value(row_set)
```

Comment:
Why does this return a `RowSet`? The return type is `article`.

Reply:
Updated!